src/util/bv_encoding.h

```diff
@@ -11,7 +11,7 @@ Author: Diffblue Ltd
 
 #include <cstddef>
 
-struct bv_encodingt
+struct bv_encodingt final
```

Comment:
I'm not sure that's necessary, I find `final` to be a bit noisy. To me `final` actually suggests the opposite of what you're saying here: This *is* part of a class hierarchy, it's just supposed to be at the bottom of it.

Reply:
You know that isn't part of a hierarchy from reading this line of code, because no base classes are specified. Having it specified as `final` also means that you are always safe to pass / return / assign instances of this class by value, without risk of slicing occurring. I do take your point about noise however. I think I may drop this commit as I don't really want to include anything in this PR, where the pros and cons could be argued either way. Suffice to say that my current habit is to make new classes/structs `final` by default, in the same way as I would make variables and member functions `const` by default.